utils/useful.py

```diff
@@ -378,41 +417,46 @@ def islicechunk(sequence: List[T], *, chunk: Optional[int] = 1) -> Generator[T,
             yield sequence[end - chunk: end]
 
 
-def text_chunker(text: str, *, width: Optional[int] = 1880, max_newline: Optional[int] = 20, wrap: Optional[bool] = True,
-                 wrap_during_chunk: Optional[bool] = True) -> List[str]:
+def text_chunker(text: str, *, width: int = 1880, max_newline: int = 20, wrap: bool = True,
+                 wrap_during_chunk: bool = True) -> List[str]:
     """Chunks a given text into a flattened list.
         text: str
             massive text that needs to be chunked
-        width: Optional[int]
+        width: int
             maximum character per chunks
-        max_newline: Optional[int]
+        max_newline: int
             maximum new line per chunks
-        wrap: Optional[bool]
+        wrap: bool
             whether to chunk before the max_newline pagination
-        wrap_during_chunk: Optional[bool]
+        wrap_during_chunk: bool
             maximum character during max_newline pagination
 
         return: List[str]
     """
     # idk i just write this long ass doc so i remember how to use it later lmao
+
+    # these recursive lists are too cursed to type, i gave up
     if wrap:
-        text = textwrap.wrap(text, width=width, replace_whitespace=False)
+        wrapped_text = textwrap.wrap(text, width=width, replace_whitespace=False)
+    else:
+        wrapped_text = [text]
 
     for i, each in enumerate(text):
         elems = each.splitlines()
         if len(elems) >= max_newline:
-            new_elems = []
+            new_elems: List[Union[str, List[str]]] = []
             for values in islicechunk(elems, chunk=20):
                 elem = "\n".join(values)
                 if wrap_during_chunk:
-                    elem = textwrap.wrap(elem, width=width, replace_whitespace=False)
-                new_elems.append(elem)
-            text[i] = new_elems
+                    new_elems.append(textwrap.wrap(elem, width=width, replace_whitespace=False))
+                else:
+                    new_elems.append(elem)
+            wrapped_text[i] = new_elems  # type: ignore
```

Comment:
This change made typing a little less hell here

Reply:
what, how, this is just unnecessary

Reply:
Fixed